packages/pwa-buildpack/src/WebpackTools/PWADevServer.js

```diff
@@ -100,79 +101,64 @@ const PWADevServer = {
-        if (id || provideSecureHost) {
-            const hostConf = {};
-            // backwards compatibility
-            if (id) {
-                const desiredDomain = id + '.' + configureHost.DEV_DOMAIN;
+        const customOriginConfig = projectConfig.section('customOrigin');
+        if (customOriginConfig.enabled) {
+            if (devServerEnvVars.host) {
+                const { host } = devServerEnvVars;
                 console.warn(
-                    debug.errorMsg(
-                        chalk.yellowBright(`
-The 'id' configuration option is deprecated and will be removed in upcoming
-releases. It has been replaced by 'provideSecureHost' configuration which can
-be configured to have the same effect as 'id'.
-
-  To create the subdomain ${desiredDomain}, use:
-    ${chalk.whiteBright(
-        `provideSecureHost: { subdomain: "${id}", addUniqueHash: false }`
-    )}
-
-  To omit the default ${
-      configureHost.DEV_DOMAIN
-  } and specify a full alternate domain, use:
-    ${chalk.whiteBright(
-        `provideSecureHost: { exactDomain: "${id}.example.dev" }`
-    )}
-  (or any other top-level domain).
-
-  ${helpText}`)
-                    )
+                    `Custom origins are enabled for this project, but the environment is overriding the custom hostname with ${JSON.stringify(
+                        devServerConfig.toEnv({ host })
+                    )}`
                 );
-
-                hostConf.addUniqueHash = false;
-                hostConf.subdomain = id;
-            } else if (provideSecureHost === true) {
-                hostConf.addUniqueHash = true;
-            } else if (typeof provideSecureHost === 'object') {
-                Object.assign(hostConf, provideSecureHost);
+                devServerConfig.host = host;
             } else {
-                throw new Error(
-                    debug.errorMsg(
-                        `Unrecognized argument to 'provideSecureHost'. Must be a boolean or an object with 'addUniqueHash', 'subdomain', and/or 'domain' properties.`
-                    )
+                const customOrigin = await configureHost(
+                    Object.assign(customOriginConfig, {
+                        interactive: false
+                    })
                 );
-            }
-            const { hostname, ports, ssl } = await configureHost(hostConf);
-
-            devServerConfig.host = hostname;
-            devServerConfig.https = ssl;
-            // workaround for https://github.com/webpack/webpack-dev-server/issues/1491
-            devServerConfig.https.spdy = {
-                protocols: ['http/1.1']
-            };
+                if (!customOrigin) {
+                    console.warn(
+                        chalk.yellowBright(
+                            'Custom origins are enabled for this project, but one has not yet been set up. Run `npx @magento/pwa-buildpack init-custom-origin <projectRoot>` to initialize a custom origin.'
```

Comment:
Consider suggesting a `yarn workspace` command.

Reply:
That would only work if they were in this repository. If they're in their own, they need to run `npx` just to get a copy of buildpack. However, we could do `yarn buildpack`.

Reply:
OK to merge it without the yarn-specific instructions for now?